core/src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1030,6 +1032,32 @@ protected QuestionDef parseControl(IFormElement parent, Element e, int controlTy
         return question;
     }
 
+    private void parseConstraintLabel(QuestionDef q, Element e) {

```

Comment:
would be nice to reduce the repeated code between this and parseQuestionLabel (particularly if we use this pattern for help/hint text)